### DISCUSSION TITLE
How to run a macOS app in this image?

### DISCUSSION THREAD

**Discussion by hangarter** [sickcodes/Docker-OSX#180] [Q&A]
Hi All,

I've started working on Unity docker images with the intention of porting it to macOS (currently it supports only linux). 
This project has a base image and a "application image" (on this case the applications could be unity editor and unity hub).

So far, I did some experiments porting the base image from linux to macos and it looks promising: https://github.com/hangarter/docker/blob/main/base-macos/Dockerfile (installing homebrew was particularly painful)

Now, to port the linux Unity Hub to macos version of the Dockerfile (https://github.com/hangarter/docker/blob/main/hub/Dockerfile) I did some tests locally where I do this:

```
ARG baseImage="base-macos"
FROM $baseImage

# Download & extract AppImage
RUN wget --no-verbose -O /tmp/UnityHubSetup.dmg "https://public-cdn.cloud.unity3d.com/hub/prod/UnityHubSetup.dmg" 
```

So the image just downloads unity's hub .dmg. 

After that I ran the image and connected via bash, extracted the .dmg with 7zip, and tried running the app that was in the extracted folder, but it says "it's an invalid executable".

Is there anything I'm missing on this docker setup?

Cheers!

**Comment by sickcodes** [comment]
Yeah there is a layer in between, Arch Linux.

Start an auto image, SSH into the Mac and download it inside the Mac as it appears you're only downloading it inside of the container.

Host -> Container -> Guest

**Comment by sickcodes** [comment]
Oh the answer to the title of the discussion is use

`-e OSX_COMMANDS='some app'`

And only for the :auto image at the moment
